src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php

```diff
@@ -104,9 +112,9 @@ public function userId(): int
         return $this->userId;
     }
 
-    public function productIdentifier(): string
+    public function identifierOrUuid(): ProductIdentifier | ProductUuid | string | null
```

Comment:
```suggestion
    public function identifierOrUuid(): ProductIdentifier | ProductUuid | null
```

If a string is provided at construct time, we can return a ProductIdentifier. Less conditions to make in the handler/validators/...